File: backend/src/main/java/com/contentplatform/backend/infrastructure/jpa/entity/MenuItemEntity.java
package com.contentplatform.backend.infrastructure.jpa.entity;

import com.contentplatform.backend.domain.value.MenuItemTarget;
import com.contentplatform.backend.domain.value.MenuItemType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.time.Instant;

@Entity
@Table(name = "menu_items")
public class MenuItemEntity {
    @Id
    @Column(name = "id", nullable = false, length = 36)
    private String id;

    @Column(name = "menu_id", nullable = false, length = 36)
    private String menuId;

    @Column(name = "parent_id", length = 36)
    private String parentId;

    @Column(name = "title", nullable = false)
    private String title;

    @Enumerated(EnumType.STRING)
    @Column(name = "item_type", nullable = false)
    private MenuItemType itemType;

    @Column(name = "reference_id", length = 36)
    private String referenceId;

    @Column(name = "url", columnDefinition = "text")
    private String url;

    @Enumerated(EnumType.STRING)
    @Column(name = "target", nullable = false)
    private MenuItemTarget target;

    @Column(name = "icon")
    private String icon;

    @Column(name = "css_class")
    private String cssClass;

    @Column(name = "sort_order", nullable = false)
    private int sortOrder;

    @Column(name = "is_visible", nullable = false)
    private boolean visible;

    private String source;
    @Column(name = "source_key")
    private String sourceKey;
    @Column(name = "managed_by", nullable = false)
    private String managedBy = "ADMIN";

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    protected MenuItemEntity() {
    }

    public MenuItemEntity(String id, String menuId, String parentId, String title, MenuItemType itemType,
                          String referenceId, String url, MenuItemTarget target, String icon, String cssClass,
                          int sortOrder, boolean visible, Instant createdAt, Instant updatedAt) {
        this.id = id;
        this.menuId = menuId;
        this.parentId = parentId;
        this.title = title;
        this.itemType = itemType;
        this.referenceId = referenceId;
        this.url = url;
        this.target = target;
        this.icon = icon;
        this.cssClass = cssClass;
        this.sortOrder = sortOrder;
        this.visible = visible;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    public void setOwnership(String source, String sourceKey, String managedBy) {
        this.source = source; this.sourceKey = sourceKey; this.managedBy = managedBy;
    }

    public String getId() { return id; }
    public String getMenuId() { return menuId; }
    public String getParentId() { return parentId; }
    public String getTitle() { return title; }
    public MenuItemType getItemType() { return itemType; }
    public String getReferenceId() { return referenceId; }
    public String getUrl() { return url; }
    public MenuItemTarget getTarget() { return target; }
    public String getIcon() { return icon; }
    public String getCssClass() { return cssClass; }
    public int getSortOrder() { return sortOrder; }
    public boolean isVisible() { return visible; }
    public Instant getCreatedAt() { return createdAt; }
    public Instant getUpdatedAt() { return updatedAt; }
    public String getSource() { return source; }
    public String getSourceKey() { return sourceKey; }
    public String getManagedBy() { return managedBy; }

    public void update(String parentId, String title, MenuItemType itemType, String referenceId, String url,
                       MenuItemTarget target, String icon, String cssClass, int sortOrder, boolean visible,
                       Instant updatedAt) {
        this.parentId = parentId;
        this.title = title;
        this.itemType = itemType;
        this.referenceId = referenceId;
        this.url = url;
        this.target = target;
        this.icon = icon;
        this.cssClass = cssClass;
        this.sortOrder = sortOrder;
        this.visible = visible;
        this.updatedAt = updatedAt;
    }
}

File: backend-nestjs/src/entities/menu-item.entity.ts
import { Column, CreateDateColumn, Entity, Index, PrimaryColumn, UpdateDateColumn } from 'typeorm';
import { MenuItemTarget, MenuItemType } from '../common/menu-types';

@Entity({ name: 'menu_items' })
@Index(['menuId', 'parentId', 'sortOrder'])
export class MenuItemEntity {
  @PrimaryColumn({ type: 'varchar', length: 36 })
  id!: string;

  @Column({ name: 'menu_id', type: 'varchar', length: 36 })
  menuId!: string;

  @Column({ name: 'parent_id', type: 'varchar', length: 36, nullable: true })
  parentId!: string | null;

  @Column({ type: 'varchar' })
  title!: string;

  @Column({ name: 'item_type', type: 'enum', enum: MenuItemType })
  itemType!: MenuItemType;

  @Column({ name: 'reference_id', type: 'varchar', length: 36, nullable: true })
  referenceId!: string | null;

  @Column({ type: 'text', nullable: true })
  url!: string | null;

  @Column({ type: 'enum', enum: MenuItemTarget, default: MenuItemTarget.SELF })
  target!: MenuItemTarget;

  @Column({ type: 'varchar', nullable: true })
  icon!: string | null;

  @Column({ name: 'css_class', type: 'varchar', nullable: true })
  cssClass!: string | null;

  @Column({ name: 'sort_order', type: 'integer', default: 0 })
  sortOrder!: number;

  @Column({ name: 'is_visible', type: 'boolean', default: true })
  isVisible!: boolean;

  @Column({ type: 'varchar', nullable: true })
  source!: string | null;

  @Column({ name: 'source_key', type: 'varchar', nullable: true })
  sourceKey!: string | null;

  @Column({ name: 'managed_by', type: 'varchar', default: 'ADMIN' })
  managedBy!: 'TENANT' | 'CMS' | 'ADMIN';

  @CreateDateColumn({ name: 'created_at', type: 'timestamptz' })
  createdAt!: Date;

  @UpdateDateColumn({ name: 'updated_at', type: 'timestamptz' })
  updatedAt!: Date;
}
